packages/neos-ui-guest-frame/src/InlineUI/NodeToolbar/Buttons/AddNode/index.js

```diff
@@ -15,8 +15,13 @@ import {neos} from '@neos-project/neos-ui-decorators';
 
     return state => {
         const focusedNodeContextPath = selectors.CR.Nodes.focusedNodePathSelector(state);
+        const getNodeByContextPathSelector = selectors.CR.Nodes.makeGetNodeByContextPathSelector(focusedNodeContextPath);
+        const focusedNode = getNodeByContextPathSelector(state);
+
+        const role = focusedNode ? (nodeTypesRegistry.hasRole(focusedNode.nodeType, 'document') ? 'document' : 'content') : null;
```

Comment:
why dont we use `nodeTypesRegistry.getRole(focusedNode.nodeType)`, as the role could not only be `document` `content`, but also

`ignored` or `contentCollection` 


see

frontendConfiguration.nodeTypeRoles

also i truly dislike nested ternaries ^^

Reply:
> why dont we use `nodeTypesRegistry.getRole(focusedNode.nodeType)`, as the role could not only be `document` `content`, but also
> 
> `ignored` or `contentCollection`

The calculation of `role` in this instance replicates the implementation of `SelectNodeType`:

https://github.com/neos/neos-ui/blob/06cf2eb4ea12ebfc07716c87d6bfe3a2383b606f/packages/neos-ui/src/Containers/Modals/SelectNodeType/index.js#L65

I agree that `nodeTypesRegistry.hasRole(focusedNode.nodeType, 'document') ? 'document' : 'content'` is one heck of a confusing statement, but it is what `SelectNodeType` is doing already. I think that fixing the semantics would be preferable. That would however require fixing it at all sites, which imho would leave the scope of this bugfix... Not sure... What's your opinion?

> also i truly dislike nested ternaries ^^

I feel similarly. Yet, if I were to resolve the nested statements into blocks, this would roughly double the footprint of this PR. I think that's not worth the trouble :shrug:

Reply:
Okay thank you for explaining this. I didn't knew we already use it like this ^^

ForGet SeMaNTIcs :D

So yes considering this, i rather like the change to be small and fix the semantics - dare i say it - later*.

*later = never